climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+    consistency, we consider an event as a whole fire season. This allows
+    consistent risk assessment across the global and over time. Events that
+    refer to a fire season have the tag 'WFseason'.
+
+    In order to perform concrete case studies or calibrate impact functions,
+    events can be displayed as single fires. In that case they have the tag
+    'WFsingle'.
+
+    Attributes:
+        date_end (np.array): integer date corresponding to the proleptic
+            Gregorian ordinal, where January 1 of year 1 has ordinal 1
+            (ordinal format of datetime library))
+
+        n_fires (np.array): number of single fires in a fire season
+
+    """
+
+    days_thres_firms = 2
+    """ Minimum number of days to consider different fires """
+
+    clus_thres_firms = 15
+    """ Clustering factor which multiplies instrument resolution """
+
+    remove_minor_fires_firms = True
+    """ removes FIRMS fires below defined theshold of entries"""
+
+    minor_fire_thres_firms = 3
+    """ number of FIRMS entries required to be considered a fire """
+
+    prop_proba = 0.21
+    """ probability of fire propagation """
+
+    max_it_propa = 500000
+    """ maximum propgation iterations for a probabilistic fire """
+
+    def __init__(self, pool=None):
+        """Empty constructor. """
+        Hazard.__init__(self, HAZ_TYPE)
+        if pool:
+            self.pool = pool
+            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
+        else:
+            self.pool = None
+
+    def set_hist_fire_FIRMS(self, csv_firms, centr_res_factor=1, centroids=None):
+        """ Parse FIRMS data and generate historical fires by temporal and spatial
+        clustering.
+        
+         This sets the attributes self.n_fires, self.date_end, in addition to all attributes required by the hazard class.
+         
+         This creates a centroids raster if centroids=None with resolution given by centr_res_factor. The centroids can be retrieved from Wildfire.centroids()
+
+        Parameters
+        ----------
+            csv_firms : pd.DataFrame or str
+                 path to csv file of FIRMS data or FIRMS data as pd.Dataframe
+                (https://firms.modaps.eosdis.nasa.gov/download/)
+            centr_res_factor : float, optional, default=1
+                resolution factor with respect to the satellite data to use
+                for centroids creation. Hence, if MODIS data (1 km res) is
+                used and centr_res_factor is set to 0.2, the grid spacing of
+                the generated centroids will equal 5 km (=1/0.2). If centroids
+                are defined, this parameter has no effect.
+            centroids : Centroids, optional
+                centroids in degrees to map data
+
+        Returns
+        -------
+            self : climada.hazard.WildFire instance
+
+        """
+        self.clear()
+
+        # read and initialize data
+        firms = self._clean_firms_csv(csv_firms)
```

Comment:
Making clear that `firms` is a dataframe will make the code much more readable.

```suggestion
        df_firms = self._clean_firms_csv(csv_firms)
```

Reply:
adjusted here - d3fb84b2